lib/dest/write-contents/write-buffer.js

```diff
@@ -1,14 +1,30 @@
 'use strict';
 
+var iconv = require('iconv-lite');
+
 var fo = require('../../file-operations');
+var getCodec = require('../../codecs');
 
 function writeBuffer(file, optResolver, onWritten) {
+  var encoding = optResolver.resolve('encoding', file);
+  var codec = getCodec(encoding);
+  if (encoding && !codec) {
+    return onWritten(new Error('Unsupported encoding: ' + encoding));
+  }
+
   var opt = {
     mode: file.stat.mode,
     flag: optResolver.resolve('flag', file),
   };
 
-  fo.writeFile(file.path, file.contents, opt, onWriteFile);
+  var contents = file.contents;
+
+  if (encoding && codec.enc !== 'utf8') {
```

Comment:
Yes, an object. The exact details seems to depend on the `encoding` used to obtain it. For instance, not all of them actually have an `enc` property with the canonized name. I've verified, though, that the `'utf8'` one does (and I only use it to check that one case).

It's all slightly weird, the `getCodec` method is part of the public API but the return value isn't documented:
https://github.com/ashtuchkin/iconv-lite/blob/master/lib/index.js#L63

I should probably add a bunch of sanity checks (in `test/integration` I guess?) to verify that iconv behaves as expected, in case future versions of it violate our assumptions.